synapse/rest/client/v2_alpha/user_directory.py

```diff
@@ -147,32 +154,72 @@ def _on_federation_query(self, args):
         if not self.hs.is_mine(user):
             raise SynapseError(400, "User is not hosted on this homeserver")
 
-        res = yield self._get_user_info(user_id)
-        defer.returnValue(res)
+        user_ids_to_info_dict = await self.store.get_info_for_users([user_id])
+        return user_ids_to_info_dict[user_id]
 
-    @defer.inlineCallbacks
-    def _get_user_info(self, user_id):
-        """Retrieve information about a given user
 
-        Args:
-            user_id (str): The User ID of a given user on this homeserver
+class UserInfoServlet(RestServlet):
+    """Bulk version of `/user/{user_id}/info` endpoint
 
-        Returns:
-            Deferred[dict]: Deactivation and expiration information for a given user
-        """
-        # Check whether user is deactivated
-        is_deactivated = yield self.store.get_user_deactivated_status(user_id)
+    GET /user/info HTTP/1.1
 
-        # Check whether user is expired
-        expiration_ts = yield self.store.get_expiration_ts_for_user(user_id)
-        is_expired = (
-            expiration_ts is not None and self.clock.time_msec() >= expiration_ts
-        )
+    Returns a dictionary of user_id to info dictionary. Supports remote users
+    """
+
+    PATTERNS = client_patterns("/user/info$")
```

Comment:
More of a nitpick, but we probably want to explicitly not serve this endpoint on anything else than `/unstable` (`releases=(0,)`) given it's not spec'd. This also applies to the non-bulk one (except if Tchap already relies on it being served on `/r0` in which case ugh fine I guess).

Reply:
Good point, I definitely agree. I've switched the new endpoint to `/unstable`.

@giomfo Do you know whether tchap uses `/_matrix/client/r0/user/info` or `_matrix/client/unstable/user/info`?

Reply:
The current implementation on ios and android was using `/r0`, but that's ok to switch on `/unstable` the corresponding code is not used in the current published mobile applications.

I just have to check this point on tchap-web side, but we will handle it. You can go for  `/unstable`

Reply:
As the single-user version is being deprecated I'll just leave it as it is for now.

But the bulk version will be `/unstable` over (`POST /_matrix/client/unstable/users/info`).